### PR TITLE
Update .NET samples for copy blob

### PR DESCRIPTION
**Pull request by pauljewellmsft**

### CODE REVIEW

**Comment by pauljewellmsft** [comment]
@jaschrep-msft Could you take a look when you get a chance? This is in our content samples repo which we pull from in dev-focused articles.

I rewrote the samples for copy and abort copy to take out the extra stuff and incorporate your suggestions. I also added a separate function to check the copy status, which I'll plan to break out in the article.

File: blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                .GetBlobContainerClient("destination-container")
+                .GetBlobClient("sample-blob.txt");
+            try
+            {
+                // Lease the source blob for the copy operation 
+                // to prevent another client from modifying it
+                BlobLeaseClient lease = sourceBlob.GetBlobLeaseClient();
+
+                // Specifying -1 for the lease interval creates an infinite lease
+                await lease.AcquireAsync(TimeSpan.FromSeconds(-1));
```
Comment (jaschrep-msft) [inline blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs:43]: Hmm. It appears we do not have a constant for this, but we *do* have `TimeSpan DataLakeLeaseClient.InfiniteLeaseDuration`. Would certainly make for better code samples. Perhaps you can file a feature request for BlobLeaseClient to contain something similar?

File: blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+                // Start the copy operation and wait for it to complete
+                CopyFromUriOperation copyOperation = await destinationBlob.StartCopyFromUriAsync(sourceBlob.Uri);
+                await AbortBlobCopyAsync(copyOperation, destinationBlob);
```
Comment (jaschrep-msft) [inline blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs:47]: I'm not sure why we are aborting the copy in this sample?

File: blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                await copyOperation.WaitForCompletionAsync();
+
+                // Release the lease on the source blob
+                await lease.ReleaseAsync();
```
Comment (jaschrep-msft) [inline blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs:51]: I would release the lease in a `finally` block.

File: blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs (hunk 1 above)
Comment (pauljewellmsft) [reply blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs:43]: Good point. I'll look into that.

File: blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs (hunk 2 above)
Comment (pauljewellmsft) [reply blobs/howto/dotnet/BlobDevGuideBlobs/CopyBlob.cs:47]: Oops, I see what you're saying now. Yes, I meant to leave the function in place but remove the function call from this spot. Thanks for pointing it out!

**Pull request merged**